lib/Database/ImportTransaction.cpp

```diff
@@ -158,13 +158,16 @@ IDCode ImportTransaction::Implementation::addModuleName(StringRef moduleName) {
 }
 
 void ImportTransaction::Implementation::addFileAssociationForProvider(IDCode provider, IDCode file, IDCode unit,
-                                                                      llvm::sys::TimeValue modTime, IDCode module, bool isSystem) {
+                                                                      llvm::sys::TimePoint<> modTime, IDCode module, bool isSystem) {
   auto &db = DBase->impl();
   auto &dbiFilesByProvider = db.getDBITimestampedFilesByProvider();
 
   auto cursor = lmdb::cursor::open(Txn, dbiFilesByProvider);
 
-  TimestampedFileForProviderData entry{file, unit, module, modTime.seconds(), modTime.nanoseconds(), isSystem};
+  auto nanos = std::chrono::duration_cast<std::chrono::nanoseconds>(modTime.time_since_epoch()).count();
+  auto secComponent = nanos / 1000000000;
+  int32_t nanoComponent = nanos % 1000000000;
```

Comment:
This changes the interpretation of any existing index data.  TimeValue's internal epoch was/is `00:00:00 Jan 1, 2000 UTC`, but this changes it to store the time using `system_clock`'s epoch, which is probably the posix epoch in 1970.

I think we need to bump `DATABASE_FORMAT_VERSION` CC @akyrtzi to verify this.

@akyrtzi since we need to bump the format anyway, is there a reason we need to split up the time into components like this?  Could we store a `std::chrono::time_point<std::chrono::system_clock, nanoseconds>` directly instead (we'd want to use that instead of `TimePoint<>` to guard against changes in llvm)?

Reply:
> @akyrtzi since we need to bump the format anyway, is there a reason we need to split up the time into components like this? Could we store a `std::chrono::time_point<std::chrono::system_clock, nanoseconds>` directly instead

I agree, we should store this as a single uint64 and bump the database format.